raiden/messages.py

```diff
@@ -292,6 +292,30 @@ class DirectTransfer(SignedMessage):
     cmdid = messages.DIRECTTRANSFER
 
     def __init__(self, identifier, nonce, token, transferred_amount, recipient, locksroot):
+        if nonce <= 0:
```

Comment:
We have the same checks in `MediatedTransfer` and in `RefundTransfer`. Instead of pasting the same checks in 3 different functions we can abstract them away in a new function and just call it in all 3 constructors.

Suggested name: `assertMessageFieldsValid()`